Source/GSAutoLayoutVFLParser.m

```diff
@@ -0,0 +1,755 @@
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import "GSAutoLayoutVFLParser.h"
+
+struct GSObjectOfPredicate
+{
+  NSNumber *priority;
+  NSView *view;
+  NSLayoutRelation relation;
+  CGFloat constant;
+};
+typedef struct GSObjectOfPredicate GSObjectOfPredicate;
+
+NSInteger const GS_DEFAULT_VIEW_SPACING = 8;
+NSInteger const GS_DEFAULT_SUPERVIEW_SPACING = 20;
+
+@implementation GSAutoLayoutVFLParser
+
+- (instancetype) initWithFormat:(NSString *)format
+                       options:(NSLayoutFormatOptions)options
+                       metrics:(NSDictionary *)metrics
+                         views:(NSDictionary *)views
+{
+  if (self = [super init])
+    {
+      if ([format length] == 0)
+        {
+          [self failParseWithMessage: @"Cannot parse an empty string"];
+        }
+
+      _views = views;
+      _metrics = metrics;
+      _options = options;
+
+      _scanner = [NSScanner scannerWithString: format];
+      _constraints = [NSMutableArray array];
+      _layoutFormatConstraints = [NSMutableArray array];
+    }
+
+  return self;
+}
+
+- (NSArray *) parse
+{
+  [self parseOrientation];
+  NSNumber *spacingConstant = [self parseLeadingSuperViewConnection];
+  NSView *previousView = nil;
+
+  while (![_scanner isAtEnd])
+    {
+      NSArray *viewConstraints = [self parseView];
+      if (_createLeadingConstraintToSuperview)
+        {
+          [self addLeadingSuperviewConstraint: spacingConstant];
+          _createLeadingConstraintToSuperview = NO;
+        }
+
+      if (previousView != nil)
+        {
+          [self addViewSpacingConstraint: spacingConstant
+                            previousView: previousView];
+          [self addFormattingConstraints: previousView];
+        }
+      [_constraints addObjectsFromArray: viewConstraints];
+
+      spacingConstant = [self parseConnection];
+      if ([_scanner scanString: @"|" intoString:nil])
+        {
+          [self addTrailingToSuperviewConstraint: spacingConstant];
+        }
+      previousView = _view;
+    }
+
+  [_constraints addObjectsFromArray: _layoutFormatConstraints];
+
+  return _constraints;
+}
+
+- (void) addFormattingConstraints:(NSView *)lastView
+{
+  BOOL hasFormatOptions = (_options & NSLayoutFormatAlignmentMask) > 0;
+  if (!hasFormatOptions)
+    {
+      return;
+    }
+  [self assertHasValidFormatLayoutOptions];
+
+  NSArray *attributes = [self layoutAttributesForLayoutFormatOptions:_options];
+  for (NSNumber *layoutAttribute in attributes)
```

Comment:
I am not sure whether our minimal gcc version supports the object for loop. Maybe you would be better off to use the FOR_IN macro, see GSFastEnumeration.h